spec.md

```diff
@@ -1798,6 +1798,45 @@ module MyModule :
    out[4] <= in
 ```
 
+## Bit-indices
+
+The bit-index expression statically refers, by index, to a particular bit, or
+slice of bits, of an expression with an integer type (`UInt`{.firrtl} or
+`SInt`{.firrtl}). The indices must be non-negative integers and cannot be equal
+to or exceed the width of the integer they index. The bit-index `x[hi:lo]`
+selects bits `hi` (most significant) through `lo` (least significant) of `x`.
+The bit-index `x[i]` selects the single bit `i`.
+
+The type of the bit-index expression `x[hi:lo]`.{firrtl} is `UInt<hi - lo +
+1>`.{firrtl} (even if `x` is an `SInt`) and the type of `x[i]`.{firrtl} is
+`UInt<1>`.{firrtl}.
+
+The bit-index can be used as a sink or source. When used as a source,
+`x[hi:lo]` is equivalent to `bits(x, hi, lo)` and `x[i]` is equivalent to
+`bits(x, i, i)`. See the `bits`{.firrtl} primitive operation
+([@sec:bit-extraction-operation]). When used as a sink, the bit-index assigns
+to only the sliced bits of the integer. If a value has multiple bit-index
+assignments, the assignments are accumulated in order according to last-connect
+semantics, in the same way as the behavior of last-connect semantics for
+aggregate types (see [@sec:last-connect-semantics]).
```

Comment:
You should clarify what happens for overlapping slices, e.g.:

```
module MyModule :
   input in: UInt<1>
   output out: UInt<4>
   out[1:0] <= UInt("b11")
   when c:
     out[2:1] <= UInt("b00")
   out[3:2] <= UInt("b11")
```

Which should result in `out=Mux(c, b1101, b1111)`.

We might also have to consider how to check whether a signal is fully assigned considering subword assignments.

Reply:
Yes I think I should also add a note about initialization, requiring that a signal be fully initialized at the bit-level if subword assignment is used with it.